source/commands/gitops/create/github.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from 'react';
+import { Box, Text } from 'ink';
+import zod from 'zod';
+import { option } from 'pastel';
+import SelectProject from '../../../components/gitops/SelectProject.js';
+import PolicyName from '../../../components/gitops/PolicyName.js';
+import SSHKey from '../../../components/gitops/SSHKey.js';
+import BranchName from '../../../components/gitops/BranchName.js';
+import Activate from '../../../components/gitops/Activate.js';
+import {
+	KEYSTORE_PERMIT_SERVICE_NAME,
+	DEFAULT_PERMIT_KEYSTORE_ACCOUNT,
+} from '../../../config.js';
+import * as keytar from 'keytar';
+
+type GitConfig = {
+	url: string;
+	main_branch_name: string;
+	credentials: {
+		auth_type: string;
+		username: string;
+		private_key: string;
+	};
+	key: string;
+};
+export const options = zod.object({
+	key: zod
+		.string()
+		.optional()
+		.describe(
+			option({
+				description:
+					'The API key for the permit Environment Organization or Project',
+				alias: 'k',
+			}),
+		),
+});
+
+type Props = {
+	options: zod.infer<typeof options>;
+};
+
+export default function GitHub({ options }: Props) {
+	const [error, setError] = useState<string>('');
+	const [projectKey, setProjectKey] = useState<string>('');
+	const [doneMessage, setDoneMessage] = useState<string>('');
+	const [gitConfig, setGitConfig] = useState<GitConfig>({
+		url: '',
+		main_branch_name: '',
+		credentials: {
+			auth_type: 'ssh',
+			username: 'git',
+			private_key: '',
+		},
+		key: '',
+	});
+	const [ApiKey, setApiKey] = useState<string>('');
+	const [state, setState] = useState<
+		| 'api_key'
+		| 'policy_name'
+		| 'ssh_key'
+		| 'branch'
+		| 'project'
+		| 'activate'
+		| 'done'
+		| 'error'
+	>('api_key');
+	useEffect(() => {
```

Comment:
Change the logic to support the AuthProvider logic (as mentioned earlier in the CR)